### PR TITLE
chore: multiple dependency updates

### PR DESCRIPTION
**Pull request by JanZachmann**
- omnect-device-service
  - bumped to 0.14.16
  - with this version a configurable outgoing message confirmation timeout is introduced. Further, the default timeout increases from 5 to 10s
- wifi-commissioning-gatt-service
  - bumped to 0.3.10 which solves audit warnings 
- azure-iot-sdk-c
  - bumped to lts_03_2024

### CODE REVIEW

**Review by mlilien — APPROVED** [review]

**Comment by JanZachmann** [comment]
@mlilien added wifi-commissioning-gatt-service to this PR

**Review by mlilien — APPROVED** [review]

File: recipes-azure-iot/azure-iot-sdk-c/azure-iot-sdk-c_1.12.0.bb [hunk 1]
```diff
@@ -4,7 +4,7 @@ HOMEPAGE = "https://github.com/Azure/azure-iot-sdk-c"
 LICENSE = "MIT"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=4283671594edec4c13aeb073c219237a"
 
-SRC_URI = "gitsm://github.com/Azure/azure-iot-sdk-c.git;branch=lts_08_2023;tag=LTS_08_2023;protocol=https"
+SRC_URI = "gitsm://github.com/Azure/azure-iot-sdk-c.git;branch=lts_03_2024;tag=LTS_03_2024;protocol=https"
```
Comment (mlilien) [inline recipes-azure-iot/azure-iot-sdk-c/azure-iot-sdk-c_1.12.0.bb:7]: filename probably has to be adapted too.

**Review by mlilien — DISMISSED** [review]

**Comment by JoergZeidler** [comment]
@JanZachmann: could we merge that PR, or are there build problems regarding the azure-sdk?

**Comment by JanZachmann** [comment]
> @JanZachmann: could we merge that PR, or are there build problems regarding the azure-sdk?

https://github.com/Azure/azure-iot-sdk-c/issues/2595

**Review by mlilien — APPROVED** [review]

**Pull request merged**